modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -127,12 +127,13 @@ inline int toCV(IE::Precision prec) {
 
 inline IE::TensorDesc toIE(const cv::Mat &mat, cv::gapi::ie::TraitAs hint) {
     const auto &sz = mat.size;
-
     // NB: For some reason RGB image is 2D image
     // (since channel component is not counted here).
-    // Note: regular 2D vectors also fall into this category
+    // In order to disnguish 2D tensor from image, have a look on model
+    // expected layout if it's 4D layout, most probably user passed mat is image.
     if (sz.dims() == 2 && hint == cv::gapi::ie::TraitAs::IMAGE)
     {
+        GAPI_Assert(sz.dims() == 2);
```

Comment:
What's the reason to have this assert here? Under this `if` it is always `true`.

Reply:
You're right, missed it